memberships/views.py

```diff
@@ -179,6 +181,24 @@ def stripe_webhook(request):
             json.loads(request.body), settings.STRIPE_SECRET_KEY
         )
         stripe_webhook = StripeWebhook()
+        if event["type"] == "invoice.payment_failed":
+            f_payment = FailedPayment.objects.create(
+                stripe_customer_id=event["data"]["object"]["customer"],
+                stripe_subscription_id=event["data"]["object"]["subscription"],
+                stripe_event_type=event["type"],
+            )
+        if event["type"] == "invoice.payment_succeeded":
```

Comment:
Could you write a test case for this `payment_succeeded` scenario?

Reply:
Of course